### PR TITLE
Check notebooks in CI `check` workflow #76

### PR DESCRIPTION
**Pull request by rlskoeser**
ref #76 

Add check for jupyter notebooks using [treon](https://github.com/ReviewNB/treon)

Update example notebook so it passes the check; use local data instead of remote.

Add Shakespeare and Company Project events dataset to use cases - both as an example and so we can reference it in the notebook.

## todo:
- [x] check if extra lxml dependency is still needed
- [x] summarize payoff / value of using undate for S&co dataset use case in the example readme

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -67,7 +69,9 @@ docs =
   sphinx<7.0.0
   sphinx_rtd_theme
   m2r2
+# NOTES:
 # pin sphinx because 7.0 currently not compatible with rtd theme
+# treon fails without lxml_html_clean (now a separate/optional package from lxml)
```
Comment (ColeDCrawford) [inline setup.cfg:74]: `treon` doesn't include `lxml_html_clean` as a dependency? Kind of odd that you need to specify it separately

File: examples/use-cases/shakespeare-and-company-project/README.md
Comment (ColeDCrawford) [inline examples/use-cases/shakespeare-and-company-project/README.md:1]: Great, this hits the dataset highlights for someone with no background on the project: what sort of events the dates are attached to, a summary of how researchers understand the dates ("business logic"), and some info about how the project has been representing them. It's clear that `undate` will be useful here, but maybe a closing statement before the citation could make it clear how `undate` improves the workflow? Just to summarize the payoff.

**Review by ColeDCrawford — APPROVED** [review]
This all looks great, the ReviewNB interface is slick. I haven't used `treon` before; that step basically makes sure you can execute all cells in the notebook in order without any errors?

**Comment by rlskoeser** [comment]
> This all looks great, the ReviewNB interface is slick. I haven't used `treon` before; that step basically makes sure you can execute all cells in the notebook in order without any errors?

Yes - if the imports don't work or syntax is wrong the check will fail.  (I think it can do more than that if you want to, but that is useful enough for my use cases).

**Comment by rlskoeser** [comment]
@ColeDCrawford thanks for the comments. I was able to address the lxml weirdness by requiring a newer version of nbconvert, which seems somewhat better.

Also thanks for pushing me more to articulate the value and potential of undate with the S&co dataset. Would you mind taking a quick look at the paragraphs I added before I merge?

**Review by ColeDCrawford — APPROVED** [review]
Awesome. The update to `nbcovert` seems good, and the extra background and references in the S&co readme make for a compelling "sell" for anyone interested in using the package.

**Pull request merged**